--- a/src-glibc/Makerules
+++ b/src-glibc/Makerules
@@ -1222,7 +1222,7 @@
 # containing an explicit list of all the needed object files when such
 # information is available.
 $(objpfx)stamp.%: files-list-% 2>/dev/null
-	echo '$$(foreach object,$(words $($(basename $(basename $(@F)))-$(suffix $(basename $@)))),$(subst .$$(%),$(suffix $(basename $@)),$(object)))' > $@T
-	mv -f $@T $@
+	echo '$$(foreach object,$(words $($(basename $(basename $(@F)))-$(suffix $(basename $@)))),$(subst .$$(%),$(suffix $(basename $@)),$(object)))' > $@T
+	rm -f $@; cp -f $@T $@; rm -f $@T

 $(objpfx)stamp.S: files-list-S 2>/dev/null
